tasks/vendoring/__init__.py

```diff
@@ -137,6 +147,127 @@ def vendor(ctx, vendor_dir):
         apply_patch(ctx, patch)
 
 
+def download_licenses(ctx, vendor_dir):
+    log('Downloading licenses')
+    tmp_dir = vendor_dir / '__tmp__'
+    ctx.run(
+        'pip download -r {0}/vendor.txt --no-binary :all: -d {1}'.format(
+            str(vendor_dir),
+            str(tmp_dir),
+        )
+    )
+    for sdist in tmp_dir.iterdir():
+        extract_license(vendor_dir, sdist)
+    drop_dir(tmp_dir)
+
+
+def extract_license(vendor_dir, sdist):
+    if sdist.suffixes[-2] == '.tar':
+        ext = sdist.suffixes[-1][1:]
+        tar = tarfile.open(sdist, mode='r:{}'.format(ext))
+        members = tar.getmembers()
+    elif sdist.suffixes[-1] == '.zip':
+        tar = zipfile.ZipFile(sdist)
+        members = tar.infolist()
+    else:
+        raise NotImplementedError('new sdist type!')
+
+    found = False
+    for member in members:
+        try:
+            name = member.name
+        except AttributeError:  # zipfile
+            name = member.filename
+        if 'LICENSE' in name or 'COPYING' in name:
+            if '/test' in name:
+                # some testing licenses in hml5lib and distlib
+                log('Ignoring {}'.format(name))
+                continue
+            found = True
+            extract_license_mamber(vendor_dir, tar, member, name)
+    if not found:
+        log('License not found in {}, fallbacking to git'.format(sdist.name))
+        license_fallback(vendor_dir, sdist, tar)
+
+
+def license_fallback(vendor_dir, sdist, tar):
+    """A very hackish way to get LICENSE from github
```

Comment:
Given the ugliness of this, I think maybe hardcoding the 2 license URLs would work better.

Reply:
Done in a fixup commit.